### PR TITLE
Add ComponentActivity support

### PR DESCRIPTION
**Pull request by anhquan291**
# Explain your changes

_Suppose there is a related issue with enough detail for a reviewer to understand your changes fully. In that case, you can omit an explanation and instead include either “Fixes #XX” or “Updates #XX” where “XX” is the issue number._

# Checklist

- [ ] I have read the [“Pull requests” section](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md#pull-requests) in the [contributing guidelines](https://github.com/kinde-oss/.github/blob/main/.github/CONTRIBUTING.md).
- [ ] I agree to the terms within the [code of conduct](https://github.com/kinde-oss/.github/blob/main/.github/CODE_OF_CONDUCT.md).

🛟 _If you need help, consider asking for advice over in the [Kinde community](https://thekindecommunity.slack.com)._

### CODE REVIEW

**Comment by DanielRivers** [comment]
@anhquan291 Could you update the PR description explaining this change and rational. 🙏

**Comment by quannguyen291** [comment]
**Explain your changes**
The changes address a compatibility issue with the Kinde SDK initialization in Android applications using Jetpack Compose. The customer faced a problem where the Kinde Login screen failed to open when the MainActivity class extended ComponentActivity instead of AppCompatActivity.

**Changes made:**
Modified KindeSDK to support ComponentActivity:
Replaced the use of AppCompatActivity with ComponentActivity in the KindeSDK class.
Updated the result handling to check for ComponentActivity.RESULT_CANCELED and ComponentActivity.RESULT_OK instead of AppCompatActivity equivalents.

**Background:**
AppCompatActivity extends FragmentActivity, which in turn extends ComponentActivity. This hierarchical relationship means that ComponentActivity is a more fundamental class that doesn't include all the additional functionality provided by AppCompatActivity and FragmentActivity.
These changes ensure that the Kinde Login functionality works correctly when the MainActivity extends ComponentActivity, accommodating the customer's need to switch from AppCompatActivity due to other library dependencies.

**Checklist:**
✅ Updated KindeSDK to support ComponentActivity.
✅ Ensured result handling uses ComponentActivity constants.
✅ Tested Kinde Login functionality with ComponentActivity.
✅ Confirmed no regressions with AppCompatActivity.

**Review by DanielRivers — APPROVED** [review]

File: sdk/src/main/kotlin/au/kinde/sdk/KindeSDK.kt [hunk 1]
```diff
@@ -64,12 +64,12 @@ class KindeSDK(
...
         val data = result.data
 
-        if (result.resultCode == AppCompatActivity.RESULT_CANCELED && data != null) {
+        if (result.resultCode == ComponentActivity.RESULT_CANCELED && data != null) {
             val ex = AuthorizationException.fromIntent(data)
             ex?.let { sdkListener.onException(LogoutException("${ex.errorDescription}")) }
         }
 
-        if (result.resultCode == AppCompatActivity.RESULT_OK && data != null) {
+        if (result.resultCode == ComponentActivity.RESULT_OK && data != null) {
```
[the remark this answers is not in the document]
Comment (DanielRivers) [reply sdk/src/main/kotlin/au/kinde/sdk/KindeSDK.kt:72]: out of scope of this change

File: sdk/src/main/kotlin/au/kinde/sdk/KindeSDK.kt [hunk 2]
```diff
@@ -64,12 +64,12 @@ class KindeSDK(
     ) { result ->
         val data = result.data
 
-        if (result.resultCode == AppCompatActivity.RESULT_CANCELED && data != null) {
+        if (result.resultCode == ComponentActivity.RESULT_CANCELED && data != null) {
```
[the remark this answers is not in the document]
Comment (DanielRivers) [reply sdk/src/main/kotlin/au/kinde/sdk/KindeSDK.kt:67]: out of scope of this change

**Pull request merged**